[pkg/stanza] Flaky test TestMatcher/Numeric_Sorting_with_grouping

Seen in https://github.com/open-telemetry/opentelemetry-collector/actions/runs/12127649470/job/33812375774?pr=11637

```
✖  fileconsumer/matcher (17ms)

=== Failed
=== FAIL: fileconsumer/matcher TestMatcher/Numeric_Sorting_with_grouping (0.00s)
    matcher_test.go:856: 
        	Error Trace:	/tmp/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/matcher/matcher_test.go:856
        	Error:      	Not equal: 
        	            	expected: []string{"err.a.123456789.log", "err.a.123456788.log", "err.a.123456787.log", "err.a.123456786.log", "err.b.123456789.log", "err.b.123456788.log"}
        	            	actual  : []string{"err.b.123456789.log", "err.b.123456788.log", "err.a.123456789.log", "err.a.123456788.log", "err.a.123456787.log", "err.a.123456786.log"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,4 @@
        	            	 ([]string) (len=6) {
        	            	+ (string) (len=19) "err.b.123456789.log",
        	            	+ (string) (len=19) "err.b.123456788.log",
        	            	  (string) (len=19) "err.a.123456789.log",
        	            	@@ -4,5 +6,3 @@
        	            	  (string) (len=19) "err.a.123456787.log",
        	            	- (string) (len=19) "err.a.123456786.log",
        	            	- (string) (len=19) "err.b.123456789.log",
        	            	- (string) (len=19) "err.b.123456788.log"
        	            	+ (string) (len=19) "err.a.123456786.log"
        	            	 }
        	Test:       	TestMatcher/Numeric_Sorting_with_grouping

=== FAIL: fileconsumer/matcher TestMatcher (0.04s)

=== FAIL: fileconsumer/matcher TestMatcher/Numeric_Sorting_with_grouping (re-run 1) (0.00s)
    matcher_test.go:856: 
        	Error Trace:	/tmp/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/matcher/matcher_test.go:856
        	Error:      	Not equal: 
        	            	expected: []string{"err.a.123456789.log", "err.a.123456788.log", "err.a.123456787.log", "err.a.123456786.log", "err.b.123456789.log", "err.b.123456788.log"}
        	            	actual  : []string{"err.b.123456789.log", "err.b.123456788.log", "err.a.123456789.log", "err.a.123456788.log", "err.a.123456787.log", "err.a.123456786.log"}
        	            	
        	            	Diff:
        	            	--- Expected
make[3]: *** [../../Makefile.Common:132: test] Error 1
make[2]: *** [Makefile:200: pkg/stanza] Error 2
make[1]: *** [Makefile:132: gotest] Error 2
make: *** [Makefile:268: check-contrib] Error 2
        	            	+++ Actual
        	            	@@ -1,2 +1,4 @@
        	            	 ([]string) (len=6) {
        	            	+ (string) (len=19) "err.b.123456789.log",
        	            	+ (string) (len=19) "err.b.123456788.log",
        	            	  (string) (len=19) "err.a.123456789.log",
        	            	@@ -4,5 +6,3 @@
        	            	  (string) (len=19) "err.a.123456787.log",
        	            	- (string) (len=19) "err.a.123456786.log",
        	            	- (string) (len=19) "err.b.123456789.log",
        	            	- (string) (len=19) "err.b.123456788.log"
        	            	+ (string) (len=19) "err.a.123456786.log"
        	            	 }
        	Test:       	TestMatcher/Numeric_Sorting_with_grouping

=== FAIL: fileconsumer/matcher TestMatcher (re-run 1) (0.01s)
```

--- a/pkg/stanza/fileconsumer/matcher/matcher_test.go
+++ b/pkg/stanza/fileconsumer/matcher/matcher_test.go
@@ -853,7 +853,7 @@ func TestMatcher(t *testing.T) {
 			} else {
 				assert.NoError(t, err)
 			}
-			assert.Equal(t, tc.expected, files)
+			assert.ElementsMatch(t, tc.expected, files)
 		})
 	}
 }

